data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/geoip/processor/BatchGeoIPDatabaseReader.java

```diff
@@ -11,13 +11,16 @@
 
 import java.net.InetAddress;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
 
 /**
  * A {@link GeoIPDatabaseReader} useful for a single batch of processing.
  */
 class BatchGeoIPDatabaseReader implements GeoIPDatabaseReader {
     private final GeoIPDatabaseReader delegate;
+    private final Map<GeoDataInput, Map<String, Object>> geoDataCache = new HashMap<>();
```

Comment:
I suppose this will persist throughout the lifecycle of the geoip processor? Shall we limit the size of cached data and have some eviction policy in place?

Reply:
@chenqi0805 , This instance `BatchGeoIPDatabaseReader` is created for each batch of events. So this map will be removed after completion of a batch. The idea here is to create a small cache for only that batch rather than creating an actual long-running cache. Part of the reason here is that we don't necessarily expect the same IPs to show up very frequently. But, when they do, they probably happen in clusters.